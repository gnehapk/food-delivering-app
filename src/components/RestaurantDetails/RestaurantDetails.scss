@import '../../color-variables.scss';

.restuarant-name {
  font-weight: bold;
  margin-top: 0.7rem;
}

.tags {
  display: flex;
  color: $color-dark-grey;
}