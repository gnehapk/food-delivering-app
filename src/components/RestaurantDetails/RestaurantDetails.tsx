import React from 'react';

import RestaurantTags from '../RestaurantTags';
import { RestaurantType } from '../RestuarantListBuilder/RestaurantListBuilder';

import './RestaurantDetails.scss';

const RestaurantDetails: React.FC<RestaurantDetailsPropsType> = (props) => (
  <div>
    <div className="restuarant-name">{props.details.name}</div>
    <div className="tags">
      <RestaurantTags tags={props.details.tags} />
      <div>&pound;&nbsp;{props.details.price}</div>
    </div>
  </div>
);

type RestaurantDetailsPropsType = {
  details: RestaurantType;
}

export default RestaurantDetails;