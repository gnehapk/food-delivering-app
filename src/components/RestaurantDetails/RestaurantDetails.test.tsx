import React from 'react';
import TestRenderer from 'react-test-renderer';
import RestaurantDetails from './RestaurantDetails';

describe('RestaurantDetails', () => {
  it('should render correctly', () => {
    const details = {
      id: '1',
      name: 'Kitchen',
      image: 'http://kitchen.png',
      url: 'http://kitchen.png',
      price: 100,
      tags: ['tag1', 'tag2'],
    };
    const tree = TestRenderer.create(<RestaurantDetails details={details} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

