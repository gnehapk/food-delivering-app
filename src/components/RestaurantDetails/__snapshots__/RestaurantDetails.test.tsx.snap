// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantDetails should render correctly 1`] = `
<div>
  <div
    className="restuarant-name"
  >
    Kitchen
  </div>
  <div
    className="tags"
  >
    <div
      className="tag-list"
    >
      <div
        className="tag"
      >
         
        tag1
      </div>
      <div
        className="tag"
      >
         
        tag2
      </div>
    </div>
    <div>
      £ 
      100
    </div>
  </div>
</div>
`;
